### PR TITLE
[15.0] base_rest: UI assets and api docs improvements

### PR DESCRIPTION
**Pull request by yankinmax**
Depends on:
- [x] https://github.com/OCA/rest-framework/pull/21
UI

### CODE REVIEW

**Comment by yankinmax** [comment]
@simahawk @JuMiSanAr @gurneyalex  UI is still not properly working. Can you pls take a look in `__manifest__.py` probably?

File: base_rest/__manifest__.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
     "assets": {
         "web.assets_backend": [
             "base_rest/static/src/scss/base_rest.scss",
+        ],
+        "web.assets_frontend": [
```
Comment (simahawk) [inline base_rest/__manifest__.py]: wrong assets https://github.com/OCA/rest-framework/blob/14.0/base_rest/views/openapi_template.xml#L41 no?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: base_rest/__manifest__.py (hunk 1 above)
Comment (simahawk) [reply base_rest/__manifest__.py]: hmm no, they are supposed to be loaded on frontend.... well, I'm too tired now. Let's check together tomorrow :stuck_out_tongue_winking_eye:

**Comment by gurneyalex** [comment]
```
$ git diff
diff --git a/base_rest/__manifest__.py b/base_rest/__manifest__.py
index b87f6d8..e4e6443 100644
--- a/base_rest/__manifest__.py
+++ b/base_rest/__manifest__.py
@@ -18,10 +18,8 @@
         "views/base_rest_view.xml",
     ],
     "assets": {
-        "web.assets_backend": [
+        "web.assets_common": [
             "base_rest/static/src/scss/base_rest.scss",
-        ],
-        "web.assets_frontend": [
             "base_rest/static/src/js/swagger_ui.js",
             "base_rest/static/lib/swagger-ui-3.51.1/swagger-ui.css",
             "base_rest/static/lib/swagger-ui-3.51.1/swagger-ui-bundle.js",

```
gives me the UI displaying correctly

**Comment by yankinmax** [comment]
@gurneyalex @simahawk UI seems to work as expected, but still have some errors in JS console:
```
Missing dependencies:     ['@web/core/commands/command_hook']
Non loaded modules:       ['@web/core/debug/debug_menu']
Debug:                    {@web/core/debug/debug_menu: {…}}
```

**Comment by simahawk** [comment]
> @gurneyalex @simahawk UI seems to work as expected, but still have some errors in JS console:
> 
> ```
> Missing dependencies:     ['@web/core/commands/command_hook']
> Non loaded modules:       ['@web/core/debug/debug_menu']
> Debug:                    {@web/core/debug/debug_menu: {…}}
> ```

at first glance I think you can ignore it.

**Comment by yankinmax** [comment]
@simahawk let's then keep it as it's

File: base_rest/__manifest__.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
         "views/base_rest_view.xml",
     ],
     "assets": {
-        "web.assets_backend": [
+        "web.assets_common": [
```
Comment (simahawk) [inline base_rest/__manifest__.py:21]: Looking again at this... IMO it's pointless and not recommendable to load these resources for ALL assets.

We should load them directly in the template or add our own assets.

**Review by simahawk — CHANGES_REQUESTED** [review]

File: base_rest/__manifest__.py (hunk 2 above)
Comment (yankinmax) [reply base_rest/__manifest__.py:21]: We can load lib assets directly in the template. It works the same. See my last fix

Comment (yankinmax) [reply base_rest/__manifest__.py:21]: @simahawk can you pls take a look at the last commit?

**Review by lmignon — APPROVED** [review]

File: base_rest/__manifest__.py (hunk 2 above)
Comment (simahawk) [reply base_rest/__manifest__.py:21]: much better!

**Comment by simahawk** [comment]
@yankinmax can you squash pls?

**Review by simahawk — APPROVED** [review]

**Pull request merged**